proto/kuksa/val/v2/types.proto

```diff
@@ -0,0 +1,239 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License 2.0 which is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+syntax = "proto3";
+
+package kuksa.val.v2;
+import "google/protobuf/timestamp.proto";
+
+option go_package = "kuksa/val/v2";
+
+message Datapoint {
+  google.protobuf.Timestamp timestamp = 1;
+
+  oneof value_state {
+    ValueFailure failure = 2;
+    Value value          = 3;
+  }
```

Comment:
@argerus @rafaeling
Why I like the "oneof":
Assuming the ValueFailure represents the reason why a "real" value is missing (maybe we should rename it to "MissingReason"?), the oneof has some advantages over two separate fields (e.g. value and value_state):
* If having a separate status with the additional state "OK" - meaning a valid value is present -, this could potentially be in mis-sync with the value field:
    * What, if status is "OK", but no value is present?
    * Or vice versa, status is not "OK", but a value is present?
* On server side it's not possible to set both in parallel (prevented by the generated code - at least in C++)
* It gives a small optimization during transmission via grpc: There either the value or the "MissingReason" is transferred.
I don't see many disadvantages:
* Code-wise (imho) it is easier, because I can better trust, that those invalid combinations of value and value_state described above cannot occur.
Just checking if a value is there and otherwise ignoring the reason why it's missing is as easy as just having a value field or having an additional value_state field. At least in C++:
```
kuksa::val::v2::Datapoint datapoint;
if (datapoint.has_value()) {
    auto value =datapoint.value();
} else {
    // handle missing value, e.g.:
    throw MissingValueException();
}
```
This C++ code would be the same - regardless if we have:
(1)
```
message Datapoint {
    google.protobuf.Timestamp timestamp = 1;
    Value value = 2;
}
```
or (2)
```
message Datapoint {
    google.protobuf.Timestamp timestamp = 1;
    Value value = 2;
    ValueState state = 3;
}
```
or (3)
```
message Datapoint {
    google.protobuf.Timestamp timestamp = 1;
    oneof value_state {
        Value value = 2;
        MissingReason reason = 3;
    }
}
```

Besides that, I think transporting additional information about a present value (like some quality information (like "outdated" or other things we discussed today), doesn't belong into an information field describing reasons why a value is missing. Those informations aren't related.